### PR TITLE
Add maintenance mode support to Doughnuts

### PR DESCRIPTION
**Pull request by JasonTulp**
Adds a maintenance mode check for the inner call within a doughnut + adds e2e test

### CODE REVIEW

**Review by zees-dev — APPROVED** [review]
Great work! LGTM! 🚀 

I guess this also needs to be ported to XRPL pallet 😅

**Comment by zees-dev** [comment]
Was also secretly hoping we could use the maintenance mode signed extension:
- `pallet_maintenance_mode::MaintenanceChecker::<T>::new(),`

Im assuming you looked into this and it wasn't straight forward 😅

File: e2e/test/Doughnuts.test.ts [hunk 1]
```diff
@@ -1516,4 +1518,143 @@ describe("Doughnuts", () => {
...
+      );
+
+      expect(section).to.equal("system");
+      expect(name).to.equal("CallFiltered");
+    }
+
+    // try remark with maintenance mode call blocked
+    {
+      await finalizeTx(alith, api.tx.sudo.sudo(api.tx.maintenanceMode.blockPallet("System", false)));
+      await finalizeTx(alith, api.tx.sudo.sudo(api.tx.maintenanceMode.blockCall("System", "remark", true)));
```
Comment (zees-dev) [inline e2e/test/Doughnuts.test.ts:1622]: Improve CI time (~4s atleast)
```suggestion
      await finalizeTx(alith, api.tx.utility.batch([
        api.tx.sudo.sudo(api.tx.maintenanceMode.blockPallet("System", false)),
        api.tx.sudo.sudo(api.tx.maintenanceMode.blockCall("System", "remark", true))
      ]));
```

File: e2e/test/Doughnuts.test.ts [hunk 2]
```diff
@@ -1516,4 +1518,143 @@ describe("Doughnuts", () => {
...
+          dispatchError.asModule
+      );
+
+      expect(section).to.equal("system");
+      expect(name).to.equal("CallFiltered");
+    }
+
+    // Disable maintenance mode
+    await finalizeTx(alith, api.tx.sudo.sudo(api.tx.maintenanceMode.blockPallet("System", false)));
+    await finalizeTx(alith, api.tx.sudo.sudo(api.tx.maintenanceMode.blockCall("System", "remark", false)));
```
Comment (zees-dev) [inline e2e/test/Doughnuts.test.ts:1658]: same here:

```suggestion
    await finalizeTx(alith, api.tx.utility.batch([
      api.tx.sudo.sudo(api.tx.maintenanceMode.blockPallet("System", false)),
      api.tx.sudo.sudo(api.tx.maintenanceMode.blockCall("System", "remark", false))
    ]));
```

**Comment by JasonTulp** [comment]
> Was also secretly hoping we could use the maintenance mode signed extension:
> 
> * `pallet_maintenance_mode::MaintenanceChecker::<T>::new(),`
> 
> Im assuming you looked into this and it wasn't straight forward 😅

Yeah the signed extension would kind of work, however would check against Doughnut.transact, not the inner call... So better to manually provide checks for the inner call

**Review by surangap — APPROVED** [review]

**Pull request merged**